### PR TITLE
[No QA]: Workspace Feed - Initial card page

### PR DESCRIPTION
**Pull request by narefyev91**
### Details
Workspace Feed - Initial card page

### Fixed Issues

$ https://github.com/Expensify/App/issues/44304
PROPOSAL:

### Tests
1. Enable Expensify card in more features for workspace
2. Open Expensify card page
3. Verify UI

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Enable Expensify card in more features for workspace
2. Open Expensify card page
3. Verify UI

### QA Steps
1. Enable Expensify card in more features for workspace
2. Open Expensify card page
3. Verify UI

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/28352309/74850c60-21d4-41fc-83c4-b6155b38f702

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/28352309/8217ea27-bfd4-4b30-8e57-d2c98b8af8b2

</details>

<details>
<summary>iOS: Native</summary>

ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios-web

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web

</details>

<details>
<summary>MacOS: Desktop</summary>

desktop

</details>

### CODE REVIEW

File: src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPage.tsx [hunk 1]
```diff
@@ -1,181 +0,0 @@
-import {useFocusEffect} from '@react-navigation/native';
```
[the remark this answers is not in the document]
Comment (narefyev91) [reply src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPage.tsx:1]: oh.. that was deleted during merge conflicts - will rollback it now

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                shouldUseScrollView
+                icon={Illustrations.HandCard}
+                headerText={translate('workspace.common.expensifyCard')}
+                route={route}
+                guidesCallTaskID={CONST.GUIDES_CALL_TASK_IDS.WORKSPACE_EXPENSIFY_CARD}
+                shouldShowOfflineIndicatorInWideScreen
+            >
+                <View style={[styles.mt3, isSmallScreenWidth ? styles.workspaceSectionMobile : styles.workspaceSection]}>
+                    <FeatureList
+                        menuItems={tripsFeatures}
```
Comment (shawnborton) [inline src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:54]: Why tripsFeatures here?

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                        title={translate('workspace.moreFeatures.expensifyCard.feed.title')}
+                        subtitle={translate('workspace.moreFeatures.expensifyCard.feed.subTitle')}
+                        ctaText={translate('workspace.moreFeatures.expensifyCard.feed.ctaTitle')}
+                        ctaAccessibilityLabel={translate('workspace.moreFeatures.expensifyCard.feed.ctaTitle')}
+                        onCtaPress={() => {}}
+                        illustrationBackgroundColor={theme.fallbackIconColor}
```
Comment (shawnborton) [inline src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:60]: I think we would use the same color used in the illustration, right?

File: assets/images/expensifyCard/cardIllustration.svg
[the remark this answers is not in the document]
Comment (narefyev91) [reply assets/images/expensifyCard/cardIllustration.svg:1]: I just exported it directly from figma

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx (hunk 3 above)
Comment (narefyev91) [reply src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:60]: yeah - i think it should be this one <rect width="680" height="220" fill="#085239"/>?
which equal fallbackIconColor: colors.green700 - am I right @shawnborton

File: assets/images/expensifyCard/cardIllustration.svg
Comment (shawnborton) [inline assets/images/expensifyCard/cardIllustration.svg:1]: Can you use this one instead? 
[emptystateheader__cascadingcards.svg.zip](https://github.com/user-attachments/files/16085372/emptystateheader__cascadingcards.svg.zip)

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx (hunk 3 above)
Comment (shawnborton) [reply src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:60]: Yup, that's correct

File: assets/images/expensifyCard/cardIllustration.svg
Comment (narefyev91) [reply assets/images/expensifyCard/cardIllustration.svg:1]: Done - 
Screenshot 2024-07-03 at 16 13 36

Comment (shawnborton) [reply assets/images/expensifyCard/cardIllustration.svg:1]: Thanks!

Comment (shawnborton) [inline assets/images/expensifyCard/cardIllustration.svg:1]: Hmm actually let me re-export that again, it seems like it didn't export correctly.

Comment (narefyev91) [reply assets/images/expensifyCard/cardIllustration.svg:1]: sure thing

Comment (shawnborton) [reply assets/images/expensifyCard/cardIllustration.svg:1]: Let's try that again, apologies! Here is a new file: [emptystateheader__cascadingcards.svg.zip](https://github.com/user-attachments/files/16085486/emptystateheader__cascadingcards.svg.zip)

Comment (narefyev91) [reply assets/images/expensifyCard/cardIllustration.svg:1]: @shawnborton looks good?
Screenshot 2024-07-03 at 16 26 05

Comment (shawnborton) [reply assets/images/expensifyCard/cardIllustration.svg:1]: Let's see the diff once you commit it and then I'll have a better idea :)

Comment (narefyev91) [reply assets/images/expensifyCard/cardIllustration.svg:1]: yeah - pushed :)

Comment (shawnborton) [reply assets/images/expensifyCard/cardIllustration.svg:1]: Okay cool, as long as it tests well, that works for me.

**Review by allgandalf — COMMENTED** [review]
Left some comments, lets get the ball rolling!!

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    return (
+        <AccessOrNotFoundWrapper
+            accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN, CONST.POLICY.ACCESS_VARIANTS.PAID]}
+            policyID={route.params.policyID}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_EXPENSIFY_CARDS_ENABLED}
+        >
+            {!cardsList && <WorkspaceCardPageEmptyState route={route} />}
+            {cardsList && <WorkspaceExpensifyCardPage route={route} />}
+        </AccessOrNotFoundWrapper>
+    );
```
[the remark this answers is not in the document]
Comment (narefyev91) [reply src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:28]: Well - each of the Component will have it's own Wrapper, ex `WorkspacePageWithSections` already has under the hood ScreenWrapper - no needs to add one more layer.
And `WorkspaceExpensifyCardPage` - has it's own as well

**Review by allgandalf — APPROVED** [review]
Tests well on all platforms, I think this should be marked as no QA (We still don't have the `BE` api implemented so QA won't be able to test this properly.

Also i think it is important to update those two comments above @narefyev91 , I'll still approve as that only changes the comments and not code.

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import WorkspaceCardPageEmptyState from './WorkspaceCardPageEmptyState';
+import WorkspaceExpensifyCardPage from './WorkspaceExpensifyCardPage';
+
+type WorkspaceCardPageFeedProps = StackScreenProps<FullScreenNavigatorParamList, typeof SCREENS.WORKSPACE.EXPENSIFY_CARD>;
+
+// TODO: remove when Onyx data is available, and pass the data to 'WorkspaceExpensifyCardPage' so that we will not make the same 'Onyx' call twice
+const cardsList: OnyxEntry<WorkspaceCardsList> = {};
+
+function WorkspaceCardPageFeed({route}: WorkspaceCardPageFeedProps) {
```
Comment (MariaHCD) [inline src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:18]: I wonder if this should be called `WorkspaceExpensifyCardPage`. Then WorkspaceExpensifyCardPage below could be called WorkspaceExpensifyCardListPage to make it more clear what each is doing

**Review by MariaHCD — COMMENTED** [review]
Looks good to me, overall. Just one question.

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx (hunk 5 above)
Comment (mountiny) [reply src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:18]: +1 to the suggestion from @MariaHCD

**Review by mountiny — DISMISSED** [review]
Just one comment same as Maria

File: src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx (hunk 5 above)
Comment (narefyev91) [reply src/pages/workspace/expensifyCard/WorkspaceCardPageFeed.tsx:18]: ok

**Review by mountiny — APPROVED** [review]
Thank you

**Review by MariaHCD — APPROVED** [review]

**Comment by shawnborton** [comment]
Sorry for missing this - but it looks like we don't have the proper 32px of padding in the empty state card on desktop. Here is what it looks like now:
CleanShot 2024-07-17 at 15 00 47@2x

But notice how the padding is different for the workspaces empty state:
CleanShot 2024-07-17 at 15 01 31@2x

**Comment by koko57** [comment]
@shawnborton I can take care of it in my PR for Empty State modal (https://github.com/Expensify/App/issues/44306)

**Comment by shawnborton** [comment]
Amazing, thanks Agata! [We have a thread in Slack with some other feedback too](https://expensify.slack.com/archives/C036QM0SLJK/p1721221346790209) in case you can handle that as well, cc @mountiny

**Comment by koko57** [comment]
@shawnborton I can't access this thread 😅 Can you @ mention me please, if it's a channel that I have an access to? 😃

**Comment by shawnborton** [comment]
Ah no worries, here are the other two comments besides the observation about modal padding:
CleanShot 2024-07-18 at 18 37 23@2x

**Comment by koko57** [comment]
@shawnborton ok, thanks! I'll be integrating the card flow with the BE, so I'll take care of these issues there 🙂 https://github.com/Expensify/App/issues/44309

**Comment by shawnborton** [comment]
Sounds good!

File: src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPageEmptyState.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            shouldUseScrollView
+            icon={Illustrations.HandCard}
+            headerText={translate('workspace.common.expensifyCard')}
+            route={route}
+            guidesCallTaskID={CONST.GUIDES_CALL_TASK_IDS.WORKSPACE_EXPENSIFY_CARD}
+            shouldShowOfflineIndicatorInWideScreen
+        >
+            <View style={[styles.mt3, shouldUseNarrowLayout ? styles.workspaceSectionMobile : styles.workspaceSection]}>
+                <FeatureList
+                    menuItems={expensifyCardFeatures}
```
Comment (Pujan92) [inline src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPageEmptyState.tsx:49]: For these screen items, the text gets truncated on a smaller screen which caused issue https://github.com/Expensify/App/issues/45901

**Pull request merged**